apps/indexer/src/utils/getVideoTracks.ts

```diff
@@ -0,0 +1,155 @@
+import { Readable } from "stream";
+import { ReadableStream } from "stream/web";
+import { Input, StreamSource, MP4, WEBM } from "mediabunny";
+
+const MAX_VIDEO_HEADER_LENGTH = 1024 * 300;
+
+enum VideoContainerType {
+  MP4 = "mp4",
+  WEBM = "webm",
+  UNKNOWN = "unknown",
+}
+
+export interface VideoTrack {
+  dimension: {
+    width: number;
+    height: number;
+  };
+  duration?: number;
+}
+
+export async function getVideoTracks(
+  response: Response,
+  requestAbortController: AbortController,
+): Promise<VideoTrack[] | undefined> {
+  const containerType = isMP4(response)
+    ? VideoContainerType.MP4
+    : isWebM(response)
+      ? VideoContainerType.WEBM
+      : VideoContainerType.UNKNOWN;
+
+  if (containerType === VideoContainerType.UNKNOWN) {
+    return;
+  }
+
+  if (!response.ok || !response.body) {
+    return;
+  }
+
+  if (requestAbortController.signal.aborted) {
+    return;
+  }
+
+  const contentLength = parseInt(response.headers.get("content-length") ?? "0");
```

Comment:
Is this really always available? I mean that you can easily send a file without `content-length` should we short circuit in that case?

Reply:
I think we can safely try to read the `MAX_VIDEO_HEADER_LENGTH` regardless of `content-length` header.

Reply:
fixed